components/sections/Send/useSendQuery.ts

```diff
@@ -0,0 +1,23 @@
+import {useMemo} from 'react';
+import {useRouter} from 'next/router';
+import {getStateFromUrlQuery} from '@utils/url/getStateFromUrlQuery';
+
+import type {TSendQuery} from 'components/sections/Send/useSendFlow';
+
+export function useSendQuery(): {initialStateFromUrl: TSendQuery | null; stateFromUrl: TSendQuery} {
+	const router = useRouter();
+	const searchParams = new URLSearchParams(router.asPath.split('?')[1]);
```

Comment:
The way to retrieve `searchParams` before the hydration instead of using `getInitialProps`